### ISSUE TITLE
CS0652 should occur for more integer inequalities

### ISSUE THREAD

**Issue by drewnoakes** [dotnet/roslyn#54140]
I noticed some code that was checking whether an unsigned integer was less than zero.

Some testing shows:

```c#
void M1(ushort n)
{
    if (n < ushort.MinValue - 1) throw new(); // CS0652 
    if (n < ushort.MinValue)     throw new();
    if (n > ushort.MaxValue)     throw new();
    if (n > ushort.MaxValue + 1) throw new(); // CS0652 
}

void M2(byte n)
{
    if (n < byte.MinValue - 1) throw new(); // CS0652 
    if (n < byte.MinValue)     throw new();
    if (n > byte.MaxValue)     throw new();
    if (n > byte.MaxValue + 1) throw new(); // CS0652 
}

void M3(int n)
{
    if (n < int.MinValue - 1) throw new(); // CS0220 
    if (n < int.MinValue)     throw new();
    if (n > int.MaxValue)     throw new();
    if (n > int.MaxValue + 1) throw new(); // CS0220
}
```

In each method I believe all four comparisons should trigger a diagnostic.

The range tests `>` and `<` are off-by-one.

**Comment by Youssef1313** [comment]
The compiler seems to be ignoring the operator when issuing this warning. It issues a diagnostic if the constant value is out-of-range for the variable type.

The fix could be done either in a warning wave or in roslyn-analyzer rule [CA1508](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1508).

**Comment by jaredpar** [comment]
The CS0652 warnings are defined as "what did the native compiler do". Even that native compiler code had a number of dead bits we found when porting to roslyn. It's not an area that we'd likely evolve here. Possible we could take this as a warning wave but the experience improvement seems a bit border line here.

It's also hard to quantify because if you look at CS0652 it's a bit of a heuristic. It for instance will never trigger when both sides are constant because the assumption of the native compiler that such code was likely generated and hence had no reason for the warning (guess they didn't believe the generated code could also generate `#pragma` suppressions). 

To do this I think we'd first need an actual design for what we wanted CS0652 to warn on.
